addons/map/XEH_postInitClient.sqf

```diff
@@ -97,35 +78,33 @@ call FUNC(determineZoom);
 
     //illumination settings
     if (GVAR(mapIllumination)) then {
-        GVAR(flashlightInUse) = "";
-        GVAR(glow) = objNull;
-
         ["ace_playerInventoryChanged", {
-            private _flashlights = [ACE_player] call FUNC(getUnitFlashlights);
-            if ((GVAR(flashlightInUse) != "") && !(GVAR(flashlightInUse) in _flashlights)) then {
-                GVAR(flashlightInUse) = "";
+            params ["_player", ""];
+            private _flashlightItems = [_player] call FUNC(getUnitFlashlights);
```

Comment:
`[_player] call FUNC(getUnitFlashlights)`is expensive, so it should only be called if it's really needed. 

I think `_flashlight isEqualTo ""` could be checked first when the inventory changes to quickly exit.